kythe/go/platform/vfs/vfs.go

```diff
@@ -106,9 +106,9 @@ func Create(ctx context.Context, path string) (io.WriteCloser, error) {
 	return Default.Create(ctx, path)
 }
 
-// CreateTemp creates a new temp file, using the Default VFS.
-func CreateTemp(ctx context.Context, dir string, pattern string) (string, error) {
-	return Default.CreateTemp(ctx, dir, pattern)
+// TempFile creates a new temp file, using the Default VFS.
+func TempFile(ctx context.Context, dir, pattern string) (*os.File, error) {
```

Comment:
Apologies, I wasn't clear in my earlier comments.

I'm not sure it's practical to return an `*os.File` here. It works fine with the implementation that falls through to the `os` package, but these shims are here because we had to support the non-standard Google filesystem types, which do not have file descriptors. 

If we do have descriptors we can get a real `*os.File`. Otherwise you might consider defining a `File` type with analogous properties, e.g.,

```go
type File interface {
   io.ReadWriteCloser
   io.Seeker
   Name() string
   // …whatever other capabilities it needs
}
```

using a subset of the capabilities of `*os.File` as a guideline. The above appears to be all we need in this case—possibly also `io.Seeker` or `io.ReaderAt` depending on whether we pass this directly to a zip writer.

Reply:
I'd already started on a more minimal NamedWriteCloser.